### PR TITLE
Autostyle layers

### PR DESCRIPTION
**Pull request by george-silva**
## Purpose

Implements an autostyle layer feature, allowing you to let GeoSync decide how to style each layer based on the layer and style names.

I added a _very minimal testing suite_ to this PR, following the tips and tricks from [this website](https://practical.li/clojure/testing/unit-testing/). We might have a lot to discuss on this, so I'm fine with removing the test suite if needed to move this forward (until we figure out best-practices, etc)

## Related Issues

Closes PLAN-180

## Submission Checklist
- [x] Included Jira issue in the PR title (e.g. `GEO1-### Did something here`)
- [x] Code passes linter rules (`clj-kondo --lint src`)
- [x] No new reflection warnings (`clojure -M:check-reflection`)

## Testing

Did manual testing with `dev-geo.planscape.org`

Following folder structure:

```
/srv/gis/planscape/test-sierra-nevada/
├── css
│   └── TotalFuelLoad_2021_scored.css
└── tier1
    └── another-folder
        └── TotalFuelLoad_2021_scored.tif
```

Produced the following results:

1. loaded style

image

2. loaded layer

image

3. association between style & layer

image

### CODE REVIEW

**Comment by lambdatronic** [comment]
# README.org
- 93: Geosync -> GeoSync
- 93: name -> names
- 93: Move `-A` option below `-O` for consistency with your `geosync.cli/cli-options`.
- 328: big number -> large number
- 329: matching declaratively -> declaratively matching
- 330: this feature -> this feature to do the work for you
- 334-336: GeoSync will fetch all existing styles in the specified
  `geoserver-workspace` and will match each newly added GIS layer with
  any style whose name appears in the end of the layer name. This
  string matching is case insensitive.
- 340-343: I think this might be easier to read as a table:
```
|------------------------------------------+-----------------------------+--------|
| Layer                                    | Style                       | Match? |
|------------------------------------------+-----------------------------+--------|
| sierra-nevada-fires                      | fires                       | Y      |
| SierraNevada_Tier2_AnnualBurnProbability | annualburnprobability       | Y      |
| SierraNevada_Tier2_AnnualBurnProbability | TIER2_ANNUALBURNPROBABILITY | Y      |
| SierraNevada_Tier2_AnnualBurnProbability | TIER1_ANNUALBURNPROBABILITY | N      |
|------------------------------------------+-----------------------------+--------|
```

# deps.edn
- 21,23: Why make these into two aliases instead of one?
- 22,26-27: Align map values into columns.

# src/geosync/cli.clj
- 214: Move `-a` option below `-s` option to match README and include a blank line between each option.
- 222: Geosync -> GeoSync
- 222: name -> names

# src/geosync/core.clj
- 209-211: Align test expressions and value expressions into columns for greater readability.
- 210: Bind `(s/lower-case layer-name)` in a `let` before the `cond` to avoid repeatedly calling this function in `filter`.
- 210: Remove the `s/lower-case` calls on the style names. See below for a more efficient solution.
- 407: Map `s/lower-case` across all the style strings here to avoid having to do it again for every GIS layer we process later.

# test/geosync/core_test.clj
- 5: `get-matching-layer-test` -> `get-matching-style-test`

File: src/geosync/cli.clj [hunk 1]
```diff
@@ -96,6 +98,10 @@
                                           ::geoserver-auth
                                           ::operation-mode))
 
+(spec/def ::geosync-config      (spec/and ::geosync-config-file
+                                          ::geoserver-auth
+                                          ::operation-mode))
+
```
Comment (Oliver-BE) [inline src/geosync/cli.clj:104]: This can be deleted, it's a duplicate of code that already exists.

**Comment by Oliver-BE** [comment]
Can you also add a small section to the README outlining what command to run to run tests in GeoSync? We can expand on this more as we fill out the testing suite.

Once you get the two above comments merged in, feel free to merge the PR. Nice work! This will be a handy feature to have and I'm happy that you took the initiative to start getting a testing framework into GeoSync. :)

**Comment by Oliver-BE** [comment]
I may be doing something wrong here, but I was unable to run the test runner locally:
```
➜  geosync git:(gsilva-autostyle) clojure -M:runner

Running tests in #{"test"}

Execution error (FileNotFoundException) at cognitect.test-runner/test (test_runner.clj:71).
Could not locate geosync/core_test__init.class, geosync/core_test.clj or geosync/core_test.cljc on classpath. Pl
ease check that namespaces with dashes use underscores in the Clojure file name.

Full report at:
/tmp/clojure-9800539956866134270.edn
```

**Pull request merged**